### PR TITLE
chore(docs): demo and doc formatting and cleanup

### PR DESCRIPTION
**Pull request by icfantv**
* add support for `interface` parsing and documenting
* add `NgbTabChangeEvent` to tab documentation
* add support for formatting in description field
* updated component API documentation to use formatting where appropriate.

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
@icfantv this is kind of big patch with many changes. Let's put it aside for alpha.0 as there are few changes in this PR that I don't think we should be doing - but I don't have enough free cycles to review it in the details.

**Comment by icfantv** [comment]
Ok.  Totally understood.  Working with that script and the typescript reflection API was like doing surgery on a bowl of spaghetti.  I do think we should, at the very least, document the event object.  I'm ok with doing this on the description for the change handler for now.

If you could please list your concerns so they can be addressed, that would be good.

**Comment by pkozlowski-opensource** [comment]
> If you could please list your concerns so they can be addressed, that would be good.

Will do as soon as I've got a minute. Knee-deep in final details before the release.

**Comment by icfantv** [comment]
> Will do as soon as I've got a minute. Knee-deep in final details before the release.

No rush.

File: src/accordion/accordion.ts [hunk 1]
```diff
@@ -73,8 +74,8 @@ export class NgbPanel {
 }
 
 /**
- * The NgbAccordion directive is a collection of panels.
- * NgbAccordion can assure that only panel can be opened at a time.
+ * The <code>NgbAccordion</code> directive is a collection of panels.
```
Comment (pkozlowski-opensource) [inline src/accordion/accordion.ts]: I would _love_ that we add a markup pass on those docs instead of adding HTML tags.

**Comment by pkozlowski-opensource** [comment]
@icfantv so, finally had time to get back to this PR. As mentioned previously it is a mixture of several distinct changes (I will _always_ prefer small, focused PRs as those are easier to review and merge):
* extraction of API docs (interface)
* formatting in  API docs (`<code>` tag)
* changes in spelling (GB -> US)

Let me tackle those one by one

=== extraction of interface API docs

To start this change has no tests so no-no for merging :-)

But formalities aside I think that we need to do deeper changes and extract info about both interfaces and classes (soon we are going to have a modal service). As such it requires more work and deserves a separate PR. Updating #222 to reflect this.

=== formatting in  API docs

IMO formatting using HTML tags is pain in a neck. We could easily add a step where we pass extracted doc through markup formatter. While doing so we should think about handling links.

All in all this is a bigger change and deserves a separate PR.

=== changes in spelling

I don't mind either way, just let's decide & stick to one form.

**Comment by pkozlowski-opensource** [comment]
IMO this one PR is in fact a collection of several bigger changes that should be tackled in separate PRs. As such I'm going to close this one (it doesn't merge cleanly anymore). Let's keep discussing needed changes here if needed.

**Pull request closed without merging**